### PR TITLE
Us 3.3 display directions between start and destination buildings

### PR DESCRIPTION
**Pull request by maddie-luu**
## Description:
This is linked to issue #15 . The addition now shows steps to navigations, and allows users to start or stop a route. It displays all the steps to get to a location for all modes of transportation offered on our app. 

Note that unit tests are not written yet for this pr, and will be completed later. There is also a UI bug where when you begin navigation, the search bar doesn't minimize, and the search bar and top direction widget slightly overlap. And when a direction starts, the path is not highlighted (bug).

## Related User Story:
#15 

## Screenshot/Demo:
US-3 3
US-3 3(2)
US-3 3(3)
US-3 3(4)
US-3 3(5)

## Type of Change:
- [x] Feature
- [ ] Bug fix
- [x] Refactor
- [ ] Documentation update
- [x] Tests
- [ ] Others

## Pull Request Checklist
- [x] Flutter CI test passed
- [ ] Sonar test passed
- [ ] Codecov test passed
- [x] Application runs without errors
- [x] Feature tested locally on a high level
- [x] Architecture conventions respected
- [x] No API keys or secrets committed
- [x] Document when necessary

## PR Review Checklist (dev)
- [x] Code is readable and well structured
- [x] No obvious bugs or edge cases missed
- [x] Tests are adequate
- [x] Naming and organization are clear
- [x] Feature works as described
- [x] No unnecessary complexity
- [x] Follows project architecture

## PR Review Checklist (QA)
- [x] Made sure to test the application fully
- [x] No bugs created by this PR

### CODE REVIEW

File: lib/services/google_directions_service.dart [hunk 1]
```diff
@@ -54,14 +118,92 @@ class GoogleDirectionsService {
...
+                    transitVehicleType ??= vehicleType;
+                    transitLineColorHex ??= line?['color'] as String?;
+                  }
+
+                  if (transitVehicleType == null) {
+                    transitVehicleType = vehicleType;
+                  }
+                }
+
+                if (mode == 'transit') {
```
Comment (enyc24) [inline lib/services/google_directions_service.dart]: better to use constants. ie: `const String transitMode = 'transit';`

File: lib/services/google_directions_service.dart [hunk 2]
```diff
@@ -26,6 +75,21 @@ class GoogleDirectionsService {
...
+      origin: origin,
+      destination: destination,
+      mode: mode,
+    );
+    return route?.points;
+  }
+
+  /// Get route details from origin to destination.
+  /// Returns decoded polyline points and leg duration text when available.
+  Future<DirectionsRouteResult?> getRouteDetails({
```
Comment (enyc24) [inline lib/services/google_directions_service.dart]: Refactor this method to reduce complexity. Lots of for loops and if statements in one method. Use helper methods where possible.

File: lib/services/location/googlemaps_livelocation.dart [hunk 3]
```diff
@@ -105,12 +105,47 @@ class _OutdoorMapPageState extends State<OutdoorMapPage> {
...
+  // tuning params (adjust as desired)
+  static const double _navZoom = 18.0;
+  static const double _navTilt =
+      60.0; // degrees — gives angled view like Google Maps
+  // Default aerial map view
+  static const double _defaultZoom = 15.0; // adjust to your normal map zoom
+  static const double _defaultTilt = 0.0;
+  static const double _defaultBearing = 0.0;
+  static const double _navBearingThresholdDegrees = 10.0;
+  static const double _navDistanceThresholdMeters = 5.0;
```
Comment (enyc24) [inline lib/services/location/googlemaps_livelocation.dart]: These two values seemed to never be used. Could be removed in that case,

File: lib/services/navigation_steps.dart [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+  }
+}
+
+/// ---- Helpers for parsing Directions API step text ----
+
+String stripHtml(String html) {
+  // Remove tags
+  final noTags = html.replaceAll(RegExp(r'<[^>]+>'), ' ');
+  // Collapse spaces
+  final collapsed = noTags.replaceAll(RegExp(r'\s+'), ' ').trim();
```
Comment (enyc24) [inline lib/services/navigation_steps.dart]: 'RegExp' is deprecated and shouldn't be used. This class will become 'final' in a future release. 'Pattern' may be a more appropriate interface to implement.

File: lib/services/navigation_steps.dart [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+  const NavigationStepsSheet({
+    super.key,
+    required this.title,
+    required this.steps,
+    this.totalDuration,
+    this.totalDistance,
+  });
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (enyc24) [inline lib/services/navigation_steps.dart]: This UI widget encompasses lots of things, therefore violating the single responsibility principle. Split them into seperate component for bettter structure.

File: lib/shared/widgets/route_preview_panel.dart [hunk 6]
```diff
@@ -56,13 +88,16 @@ class _RoutePreviewPanelState extends State<RoutePreviewPanel> {
 
     _originFocus.addListener(() {
       setState(() {
-        _showOriginSuggestions = _originFocus.hasFocus && widget.originSuggestions.isNotEmpty;
+        _showOriginSuggestions =
```
Comment (enyc24) [inline lib/shared/widgets/route_preview_panel.dart]: This is used multiple times throughout the code. Could use a helper method instead.

File: lib/services/location/googlemaps_livelocation.dart [hunk 7]
```diff
@@ -351,16 +393,16 @@ class _OutdoorMapPageState extends State<OutdoorMapPage> {
...
 
     // Fetch the initial route
-    await _fetchRoute();
+    await _fetchRoutesAndDurations();
   }
 
-  Future<void> _fetchRoute() async {
+  Future<void> _fetchRoutesAndDurations() async {
     final origin = _routeOrigin;
     final destination = _routeDestination;
```
[the remark this answers is not in the document]
Comment (maddie-luu) [reply lib/services/location/googlemaps_livelocation.dart]: No, because this is a function that gets executed whenever we need to fetch a route to launch our navigation. A null origin and destination is already handled gracefully in a conditional statement at line 394.
